lib/rules/no-lazy-ends.ts

```diff
@@ -64,13 +73,27 @@ export default createRule("no-lazy-ends", {
         type: "problem",
     },
     create(context) {
+        const ignorePartial = context.options[0]?.ignorePartial ?? true
+
         /**
          * Create visitor
          */
         function createVisitor({
             node,
             getRegexpLocation,
+            getUsageOfPattern,
         }: RegExpContext): RegExpVisitor.Handlers {
+            if (ignorePartial) {
+                const usageOfPattern = getUsageOfPattern()
+                if (
+                    usageOfPattern === UsageOfPattern.partial ||
+                    usageOfPattern === UsageOfPattern.mixed
```

Comment:
Should we [also make this conservative](https://github.com/ota-meshi/eslint-plugin-regexp/pull/215#discussion_r642066441)?

I think that `no-useless-non-capturing-group` should be conservative because it's auto-fixable, so false positives can cause harm directly. But `no-lazy-ends` isn't auto-fixable. The user has to decide each case. Maybe false positives are more acceptable here?